{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as graph\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 7min 37s, sys: 3min 20s, total: 10min 58s\n",
      "Wall time: 12min 3s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "df = pd.read_table('/data4/Ant_genomes/invictus-codon.table')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AF</th>\n",
       "      <th>AC</th>\n",
       "      <th>AN</th>\n",
       "      <th>DP</th>\n",
       "      <th>EFF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>379</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>71.69</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>142</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.379A&gt;C|||||||C)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>380</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>72.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>143</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.380A&gt;C|||||||C)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>381</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>70.49</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>141</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.381A&gt;T|||||||T)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>385</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>76.79</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>140</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.385T&gt;A|||||||A)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>386</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>76.04</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>139</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.386C&gt;G|||||||G)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            CHROM  POS REF ALT   QUAL FILTER  AF  AC  AN   DP  \\\n",
       "0  NW_020456309.1  379   A   C  71.69   PASS NaN NaN  40  142   \n",
       "1  NW_020456309.1  380   A   C  72.36   PASS NaN NaN  40  143   \n",
       "2  NW_020456309.1  381   A   T  70.49   PASS NaN NaN  40  141   \n",
       "3  NW_020456309.1  385   T   A  76.79   PASS NaN NaN  40  140   \n",
       "4  NW_020456309.1  386   C   G  76.04   PASS NaN NaN  40  139   \n",
       "\n",
       "                                              EFF  \n",
       "0  intergenic_region(MODIFIER|||n.379A>C|||||||C)  \n",
       "1  intergenic_region(MODIFIER|||n.380A>C|||||||C)  \n",
       "2  intergenic_region(MODIFIER|||n.381A>T|||||||T)  \n",
       "3  intergenic_region(MODIFIER|||n.385T>A|||||||A)  \n",
       "4  intergenic_region(MODIFIER|||n.386C>G|||||||G)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "for allele_column in ['AF', 'AC', 'AN']:\n",
    "    df[allele_column].fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate folded allele frequency\n",
    "\n",
    "is_af_less_50 = df['AF'] <= 0.5\n",
    "\n",
    "df['cal_af'] = np.where(is_af_less_50, df['AC'], df['AN'] - df['AC'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = df.sample(n=10000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AF</th>\n",
       "      <th>AC</th>\n",
       "      <th>AN</th>\n",
       "      <th>DP</th>\n",
       "      <th>EFF</th>\n",
       "      <th>cal_af</th>\n",
       "      <th>effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>379</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>71.69</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>142</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.379A&gt;C|||||||C)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>380</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>72.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>143</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.380A&gt;C|||||||C)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>381</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>70.49</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>141</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.381A&gt;T|||||||T)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>385</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>76.79</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>140</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.385T&gt;A|||||||A)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>386</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>76.04</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>139</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.386C&gt;G|||||||G)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            CHROM  POS REF ALT   QUAL FILTER   AF   AC  AN   DP  \\\n",
       "0  NW_020456309.1  379   A   C  71.69   PASS  0.0  0.0  40  142   \n",
       "1  NW_020456309.1  380   A   C  72.36   PASS  0.0  0.0  40  143   \n",
       "2  NW_020456309.1  381   A   T  70.49   PASS  0.0  0.0  40  141   \n",
       "3  NW_020456309.1  385   T   A  76.79   PASS  0.0  0.0  40  140   \n",
       "4  NW_020456309.1  386   C   G  76.04   PASS  0.0  0.0  40  139   \n",
       "\n",
       "                                              EFF  cal_af             effect  \n",
       "0  intergenic_region(MODIFIER|||n.379A>C|||||||C)     0.0  intergenic_region  \n",
       "1  intergenic_region(MODIFIER|||n.380A>C|||||||C)     0.0  intergenic_region  \n",
       "2  intergenic_region(MODIFIER|||n.381A>T|||||||T)     0.0  intergenic_region  \n",
       "3  intergenic_region(MODIFIER|||n.385T>A|||||||A)     0.0  intergenic_region  \n",
       "4  intergenic_region(MODIFIER|||n.386C>G|||||||G)     0.0  intergenic_region  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['effect'] = df['EFF'].str.split('(').str.get(0)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AF</th>\n",
       "      <th>AC</th>\n",
       "      <th>AN</th>\n",
       "      <th>DP</th>\n",
       "      <th>EFF</th>\n",
       "      <th>cal_af</th>\n",
       "      <th>effect</th>\n",
       "      <th>genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>379</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>71.69</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>142</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.379A&gt;C|||||||C)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>380</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>72.36</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>143</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.380A&gt;C|||||||C)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>381</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>70.49</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>141</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.381A&gt;T|||||||T)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>385</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>76.79</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>140</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.385T&gt;A|||||||A)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>386</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>76.04</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>139</td>\n",
       "      <td>intergenic_region(MODIFIER|||n.386C&gt;G|||||||G)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            CHROM  POS REF ALT   QUAL FILTER   AF   AC  AN   DP  \\\n",
       "0  NW_020456309.1  379   A   C  71.69   PASS  0.0  0.0  40  142   \n",
       "1  NW_020456309.1  380   A   C  72.36   PASS  0.0  0.0  40  143   \n",
       "2  NW_020456309.1  381   A   T  70.49   PASS  0.0  0.0  40  141   \n",
       "3  NW_020456309.1  385   T   A  76.79   PASS  0.0  0.0  40  140   \n",
       "4  NW_020456309.1  386   C   G  76.04   PASS  0.0  0.0  40  139   \n",
       "\n",
       "                                              EFF  cal_af             effect  \\\n",
       "0  intergenic_region(MODIFIER|||n.379A>C|||||||C)     0.0  intergenic_region   \n",
       "1  intergenic_region(MODIFIER|||n.380A>C|||||||C)     0.0  intergenic_region   \n",
       "2  intergenic_region(MODIFIER|||n.381A>T|||||||T)     0.0  intergenic_region   \n",
       "3  intergenic_region(MODIFIER|||n.385T>A|||||||A)     0.0  intergenic_region   \n",
       "4  intergenic_region(MODIFIER|||n.386C>G|||||||G)     0.0  intergenic_region   \n",
       "\n",
       "  genes  \n",
       "0   NaN  \n",
       "1   NaN  \n",
       "2   NaN  \n",
       "3   NaN  \n",
       "4   NaN  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['genes'] = df['EFF'].str.extract('(XM_[0-9]{9}.[0-9]{1})')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>QUAL</th>\n",
       "      <th>AF</th>\n",
       "      <th>AC</th>\n",
       "      <th>AN</th>\n",
       "      <th>cal_af</th>\n",
       "      <th>effect</th>\n",
       "      <th>genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>71.69</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>72.36</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>70.49</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>76.79</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>76.04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    QUAL   AF   AC  AN  cal_af             effect genes\n",
       "0  71.69  0.0  0.0  40     0.0  intergenic_region   NaN\n",
       "1  72.36  0.0  0.0  40     0.0  intergenic_region   NaN\n",
       "2  70.49  0.0  0.0  40     0.0  intergenic_region   NaN\n",
       "3  76.79  0.0  0.0  40     0.0  intergenic_region   NaN\n",
       "4  76.04  0.0  0.0  40     0.0  intergenic_region   NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.drop(columns=['CHROM', 'POS', 'FILTER', 'REF', 'ALT', 'DP', 'EFF'], \n",
    "           inplace=True)\n",
    "display(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([nan, 'XM_011162030.2', 'XM_011162031.2', ..., 'XM_011175916.2',\n",
       "       'XM_026141870.1', 'XM_026141872.1'], dtype=object)"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['genes'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('/data4/Ant_genomes/all_ant_invictus_simple.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/arshad/miniconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n",
      "/home/arshad/miniconda3/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/arshad/miniconda3/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "# synonymous_snps = df[df['effect'].str.contains('synonymous_variant', na=False)]\n",
    "synonymous_snps['codons'] = synonymous_snps['EFF'].str.split('(').str.get(1).str.split('|').str.get(2)\n",
    "synonymous_snps['OG-codon'] = synonymous_snps['codons'].str.split('/').str.get(0)\n",
    "synonymous_snps['new-codon'] = synonymous_snps['codons'].str.split('/').str.get(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2380382\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AF</th>\n",
       "      <th>AC</th>\n",
       "      <th>AN</th>\n",
       "      <th>DP</th>\n",
       "      <th>EFF</th>\n",
       "      <th>cal_af</th>\n",
       "      <th>effect</th>\n",
       "      <th>genes</th>\n",
       "      <th>codons</th>\n",
       "      <th>OG-codon</th>\n",
       "      <th>new-codon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1152</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>5846</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>73.27</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>107</td>\n",
       "      <td>synonymous_variant(LOW|SILENT|ggC/ggT|p.Gly5Gly/c.15C&gt;T|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|T)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>XM_011162030.2</td>\n",
       "      <td>ggC/ggT</td>\n",
       "      <td>ggC</td>\n",
       "      <td>ggT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1158</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>5852</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>67.79</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>108</td>\n",
       "      <td>synonymous_variant(LOW|SILENT|ttC/ttT|p.Phe7Phe/c.21C&gt;T|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|T)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>XM_011162030.2</td>\n",
       "      <td>ttC/ttT</td>\n",
       "      <td>ttC</td>\n",
       "      <td>ttT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1160</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>5915</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>64.82</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>105</td>\n",
       "      <td>synonymous_variant(LOW|SILENT|gcC/gcG|p.Ala28Ala/c.84C&gt;G|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|G)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>XM_011162030.2</td>\n",
       "      <td>gcC/gcG</td>\n",
       "      <td>gcC</td>\n",
       "      <td>gcG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1162</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>6071</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>71.56</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>112</td>\n",
       "      <td>synonymous_variant(LOW|SILENT|gcG/gcA|p.Ala80Ala/c.240G&gt;A|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|A)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>XM_011162030.2</td>\n",
       "      <td>gcG/gcA</td>\n",
       "      <td>gcG</td>\n",
       "      <td>gcA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1166</th>\n",
       "      <td>NW_020456309.1</td>\n",
       "      <td>6075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>73.44</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>113</td>\n",
       "      <td>synonymous_variant(LOW|SILENT|Ctg/Ttg|p.Leu82Leu/c.244C&gt;T|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|T)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>XM_011162030.2</td>\n",
       "      <td>Ctg/Ttg</td>\n",
       "      <td>Ctg</td>\n",
       "      <td>Ttg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               CHROM   POS REF ALT   QUAL FILTER   AF   AC  AN   DP  \\\n",
       "1152  NW_020456309.1  5846   C   T  73.27   PASS  0.0  0.0  40  107   \n",
       "1158  NW_020456309.1  5852   C   T  67.79   PASS  0.0  0.0  40  108   \n",
       "1160  NW_020456309.1  5915   C   G  64.82   PASS  0.0  0.0  40  105   \n",
       "1162  NW_020456309.1  6071   G   A  71.56   PASS  0.0  0.0  40  112   \n",
       "1166  NW_020456309.1  6075   C   T  73.44   PASS  0.0  0.0  40  113   \n",
       "\n",
       "                                                                                                                       EFF  \\\n",
       "1152    synonymous_variant(LOW|SILENT|ggC/ggT|p.Gly5Gly/c.15C>T|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|T)   \n",
       "1158    synonymous_variant(LOW|SILENT|ttC/ttT|p.Phe7Phe/c.21C>T|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|T)   \n",
       "1160   synonymous_variant(LOW|SILENT|gcC/gcG|p.Ala28Ala/c.84C>G|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|G)   \n",
       "1162  synonymous_variant(LOW|SILENT|gcG/gcA|p.Ala80Ala/c.240G>A|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|A)   \n",
       "1166  synonymous_variant(LOW|SILENT|Ctg/Ttg|p.Leu82Leu/c.244C>T|590|LOC105196222|protein_coding|CODING|XM_011162030.2|2|T)   \n",
       "\n",
       "      cal_af              effect           genes   codons OG-codon new-codon  \n",
       "1152     0.0  synonymous_variant  XM_011162030.2  ggC/ggT      ggC       ggT  \n",
       "1158     0.0  synonymous_variant  XM_011162030.2  ttC/ttT      ttC       ttT  \n",
       "1160     0.0  synonymous_variant  XM_011162030.2  gcC/gcG      gcC       gcG  \n",
       "1162     0.0  synonymous_variant  XM_011162030.2  gcG/gcA      gcG       gcA  \n",
       "1166     0.0  synonymous_variant  XM_011162030.2  Ctg/Ttg      Ctg       Ttg  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(len(synonymous_snps))\n",
    "display(synonymous_snps.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "synon = synonymous_snps.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "synon.drop(columns=['CHROM', 'POS', 'FILTER', 'REF', 'ALT', 'DP', 'EFF'], \n",
    "           inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "# synon.to_csv('/data4/Ant_genomes/synonymous_ant_invictus_simple.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
